carma-messenger-core/cpp_message/src/MobilityPath_Message.cpp

```diff
@@ -0,0 +1,318 @@
+            }
+               header.timestamp=timestamp;
+
+            output.header=header;
+            
+            //Trajectory
+            //contains location and offset-decode to store in trajectory values
+            cav_msgs::LocationECEF location;
+            long tmp=message->value.choice.TestMessage02.body.location.ecefX;
+            if(tmp>LOCATION_MAX || tmp<LOCATION_MIN){
+                ROS_WARN_STREAM("Location ecefX is out of range");
+                //return boost::optional<cav_msgs::MobilityPath>{};
+            }
+            location.ecef_x=tmp;
+
+            tmp=message->value.choice.TestMessage02.body.location.ecefY;
+            if(tmp>LOCATION_MAX || tmp<LOCATION_MIN){
+                ROS_WARN_STREAM("Location ecefY is out of range");
+                //return boost::optional<cav_msgs::MobilityPath>{};
+            }
+            location.ecef_y=tmp;
+
+            tmp=message->value.choice.TestMessage02.body.location.ecefZ;
+            if(tmp>LOCATION_MAX_Z || tmp<LOCATION_MIN_Z){
+                ROS_WARN_STREAM("Location ecefZ is out of range");
+                //return boost::optional<cav_msgs::MobilityPath>{};
+            }
+            location.ecef_z=tmp;
+            
+            //convert location timestamp from string in asn1 to uint64 for ros message
+            str_len=message->value.choice.TestMessage02.body.location.timestamp.size;
+            uint64_t location_timestamp=0;
+            for(auto i=0;i<str_len;i++){
+                location_timestamp*=10;
+                location_timestamp+=int(message->value.choice.TestMessage02.body.location.timestamp.buf[i])-'0';
+            }
+            location.timestamp=location_timestamp;
+
+            trajectory.location=location;
+            
+            //Trajectory-offset
+            auto offset_count=message->value.choice.TestMessage02.body.trajectory.list.count;
+            
+            if(offset_count>MAX_POINTS_IN_MESSAGE){
+            ROS_WARN_STREAM("offset count greater than 60.");
+            return boost::optional<cav_msgs::MobilityPath>{};
+            } 
+
+            for(auto i=0;i<offset_count;i++){
+                cav_msgs::LocationOffsetECEF Offsets;
+                Offsets.offset_x=message->value.choice.TestMessage02.body.trajectory.list.array[i]->offsetX;
+                if(Offsets.offset_x<OFFSET_MIN || Offsets.offset_x>OFFSET_MAX)Offsets.offset_x=OFFSET_UNAVAILABLE;
+       
+                Offsets.offset_y=message->value.choice.TestMessage02.body.trajectory.list.array[i]->offsetY;
+                if(Offsets.offset_y<OFFSET_MIN || Offsets.offset_y>OFFSET_MAX)Offsets.offset_y=OFFSET_UNAVAILABLE;
+
+                Offsets.offset_z=message->value.choice.TestMessage02.body.trajectory.list.array[i]->offsetZ;
+                if(Offsets.offset_z<OFFSET_MIN || Offsets.offset_z>OFFSET_MAX)Offsets.offset_z=OFFSET_UNAVAILABLE;
+                trajectory.offsets.push_back(Offsets);
```

Comment:
For the three if statements above this please put the content of the statement on line down since the same line is making it very long
such as
```
if (...)
  do_something();
```